app/lib/features/news/widgets/news_item.dart

```diff
@@ -25,112 +26,145 @@ class NewsItem extends ConsumerWidget {
   });
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final slides = news.slides().toList();
+  ConsumerState<NewsItem> createState() => _NewsItemState();
+}
+
+class _NewsItemState extends ConsumerState<NewsItem> {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) {
+        ref.invalidate(newsSlideIndexProvider);
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final roomId = widget.news.roomId().toString();
+    final space = ref.watch(briefSpaceItemProvider(roomId));
+    final slides = widget.news.slides().toList();
+    final slideIndex = ref.watch(newsSlideIndexProvider.notifier).state;
     final bgColor = convertColor(
-      news.colors()?.background(),
+      widget.news.colors()?.background(),
       Theme.of(context).colorScheme.background,
     );
     final fgColor = convertColor(
-      news.colors()?.color(),
+      widget.news.colors()?.color(),
       Theme.of(context).colorScheme.onPrimary,
     );
 
-    return PageView.builder(
-      scrollDirection: Axis.horizontal,
-      itemCount: slides.length,
-      itemBuilder: (context, idx) {
-        final slideType = slides[idx].typeStr();
-        switch (slideType) {
-          case 'image':
-            return RegularSlide(
-              news: news,
-              index: index,
-              bgColor: bgColor,
-              fgColor: fgColor,
-              child: ImageSlide(slide: slides[idx]),
-            );
-
-          case 'video':
-            return RegularSlide(
-              news: news,
-              index: index,
-              bgColor: bgColor,
-              fgColor: fgColor,
-              child: const Expanded(
-                child: Center(
-                  child: Text('video slides not yet supported'),
+    return Stack(
+      children: [
+        PageView.builder(
+          scrollDirection: Axis.horizontal,
+          itemCount: slides.length,
+          onPageChanged: (page) {
+            ref.watch(newsSlideIndexProvider.notifier).state = page;
+            setState(() {});
```

Comment:
Why ```setState(() {})``` is needed ? we are internally changing state of notifier which triggers UI build on its own.